### PR TITLE
second part of task solved

### PR DESCRIPTION
**Pull request by Baldinini**

### CODE REVIEW

**Review by yurwing — APPROVED** [review]

File: src/main/java/mate/academy/dao/impl/CarDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+public class CarDaoImpl implements CarDao {
+
+    @Override
+    public Car create(Car car) {
+        Storage.addCar(car);
+        return car;
+    }
+
+    @Override
+    public Car get(Long id) {
```
Comment (Sofasmile) [inline src/main/java/mate/academy/dao/impl/CarDaoImpl.java]:
```suggestion
    public Optional<Car> get(Long id) {
```

File: src/main/java/mate/academy/dao/impl/CarDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    public List<Car> getAll() {
+        return Storage.cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        IntStream.range(0, Storage.cars.size())
+                .filter(i -> Storage.cars.get(i).getId().equals(car.getId()))
+                .forEach(i -> Storage.cars.set(i, car));
+        return null;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/dao/impl/CarDaoImpl.java]: return null is bad practice
```suggestion
        return car;
```

File: src/main/java/mate/academy/dao/impl/CarDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        Driver driver = Storage.drivers.stream()
+                .filter(d -> d.getId().equals(driverId))
+                .findFirst()
+                .orElse(null);
+        return Storage.cars.stream()
+                .filter(c -> c.getDrivers() != null && c.getDrivers().contains(driver))
```
Comment (Sofasmile) [inline src/main/java/mate/academy/dao/impl/CarDaoImpl.java]:
```suggestion
                .filter(c -> c.getDrivers().stream()
                        .anyMatch(d -> d.getId().equals(driverId)))
```

File: src/main/java/mate/academy/dao/impl/DriverDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+@Dao
+public class DriverDaoImpl implements DriverDao {
+    @Override
+    public Driver create(Driver driver) {
+        Storage.addDriver(driver);
+        return driver;
+    }
+
+    @Override
+    public Driver get(Long id) {
```
Comment (Sofasmile) [inline src/main/java/mate/academy/dao/impl/DriverDaoImpl.java]: the same

File: src/main/java/mate/academy/service/impl/CarServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    @Override
+    public boolean delete(Long id) {
+        return carDao.delete(id);
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        if (!driverDao.getAll().contains(driver)) {
+            driverDao.create(driver);
+        }
```
Comment (Sofasmile) [inline src/main/java/mate/academy/service/impl/CarServiceImpl.java]: let's assume that the driver exists in a DB therefore we do not need to create it

File: src/main/java/mate/academy/service/impl/CarServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        List<Driver> drivers = car.getDrivers() != null ? car.getDrivers() : new ArrayList<>();
```
Comment (Sofasmile) [inline src/main/java/mate/academy/service/impl/CarServiceImpl.java]: to fix if let's create empty ArrayList of drivers in the constructor of car

File: src/main/java/mate/academy/service/impl/CarServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        carDao.update(car);
+    }
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
+        if (car.getDrivers() != null && car.getDrivers().contains(driver)) {
+            car.getDrivers().remove(driver);
+            carDao.update(car);
+        } else {
+            System.out.println("Car doesn't have this driver: " + driver);
```
Comment (Sofasmile) [inline src/main/java/mate/academy/service/impl/CarServiceImpl.java]: sout is a bad practice, let's remove all checking

File: src/main/java/mate/academy/service/impl/ManufacturerServiceImpl.java [hunk 8]
```diff
@@ -18,7 +19,7 @@ public Manufacturer create(Manufacturer manufacturer) {
 
     @Override
     public Manufacturer get(Long id) {
-        return manufacturerDao.get(id).get();
+        return manufacturerDao.get(id).orElse(null);
```
Comment (Sofasmile) [inline src/main/java/mate/academy/service/impl/ManufacturerServiceImpl.java]:
```suggestion
        return manufacturerDao.get(id).get();
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/service/impl/CarServiceImpl.java [hunk 9]
```diff
@@ -44,23 +40,16 @@ public boolean delete(Long id) {
...
 
     @Override
     public void removeDriverFromCar(Driver driver, Car car) {
-        if (car.getDrivers() != null && car.getDrivers().contains(driver)) {
-            car.getDrivers().remove(driver);
-            carDao.update(car);
-        } else {
-            System.out.println("Car doesn't have this driver: " + driver);
-        }
+        car.getDrivers().removeIf(d -> car.getDrivers().contains(driver));
```
Comment (Sofasmile) [inline src/main/java/mate/academy/service/impl/CarServiceImpl.java:51]:
```suggestion
        car.getDrivers().remove(driver);
```

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**